provider/source/src/decimal/mod.rs

```diff
@@ -90,4 +100,49 @@ impl SourceDataProvider {
             })
             .collect())
     }
+
+    /// Produce DataIdentifier's for all *used* numbering systems in the form und/<numsys>
+    fn iter_ids_for_used_numbers(&self) -> Result<HashSet<DataIdentifierCow<'static>>, DataError> {
+        Ok(self
+            .cldr()?
+            .numbers()
+            .list_locales()?
+            .flat_map(|locale| {
+                self.get_supported_numsys_for_langid(&locale, false)
+                    .expect("All languages from list_locales should be present")
+                    .into_iter()
+                    .map(move |nsname| {
+                        DataIdentifierBorrowed::for_marker_attributes_and_locale(
+                            DataMarkerAttributes::try_from_str(&nsname).unwrap(),
+                            &locale!("und").into(),
+                        )
+                        .into_owned()
+                    })
+            })
+            .collect())
+    }
+
+    /// Produce DataIdentifier's for all digit-based numbering systems in the form und/<numsys>
+    #[allow(unused)] // TODO configurable
```

Comment:
Nit: Add an issue number. I don't expect this to be resolved in 2.0 and it can probably wait until @robertbastian is back.